pkg/orchestration/wiring/kubeingress/plugin_test.go

```diff
@@ -5,14 +5,94 @@ import (
 
 	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
 	"github.com/atlassian/voyager"
+	"github.com/atlassian/voyager/pkg/apis/orchestration/v1"
 	"github.com/atlassian/voyager/pkg/k8s"
 	"github.com/atlassian/voyager/pkg/orchestration/wiring/k8scompute/api"
 	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringplugin"
 	"github.com/stretchr/testify/assert"
 	apps_v1 "k8s.io/api/apps/v1"
+	ext_v1b1 "k8s.io/api/extensions/v1beta1"
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+func TestBuildingIngressResource(t *testing.T) {
+	t.Parallel()
+
+	var serviceName smith_v1.ResourceName = "serviceName"
+
+	validOutput := wiringplugin.WiredSmithResource{
+		SmithResource: smith_v1.Resource{
+			Name: "somename-ingress",
+			References: []smith_v1.Reference{
+				{
+					Resource: serviceName,
+				},
+			},
+			Spec: smith_v1.ResourceSpec{
+				Object: &ext_v1b1.Ingress{
+					TypeMeta: meta_v1.TypeMeta{
+						Kind:       k8s.IngressKind,
+						APIVersion: ext_v1b1.SchemeGroupVersion.String(),
+					},
+					ObjectMeta: meta_v1.ObjectMeta{
+						Name: "somename-ingress",
```

Comment:
... and the name of the kube object. They do match here but they don't have to. And things will not work if we change one but not the other. But they should still work.